heapsort.java

```diff
@@ -0,0 +1,74 @@
+// Java program for implementation of Heap Sort 
+public class HeapSort 
+{ 
+	public void sort(int arr[]) 
+	{ 
+		int n = arr.length; 
+
+		// Build heap (rearrange array) 
+		for (int i = n / 2 - 1; i >= 0; i--) 
+			heapify(arr, n, i); 
+
+		// One by one extract an element from heap 
+		for (int i=n-1; i>0; i--) 
+		{ 
+			// Move current root to end 
+			int temp = arr[0]; 
+			arr[0] = arr[i]; 
+			arr[i] = temp; 
+
+			// call max heapify on the reduced heap 
+			heapify(arr, i, 0); 
+		} 
+	} 
+
+	// To heapify a subtree rooted with node i which is 
+	// an index in arr[]. n is size of heap 
+	void heapify(int arr[], int n, int i) 

```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6319142)